Add support for unit type numbered

Joxit opened the pull request:
Hi there !

I added some support for `unit` :tada: 

**Classifications**
- `UnitTypeClassification`: public classifications for types like `unit`, `apt`...
- `UnitClassification`: public classification for unit numbers

**Classifiers**
- `UnitTypeClassifier`: classify a UnityType thanks to libpostal `unit_types_numbered.txt` (I removed `stop` and `shop`)
- `UnitClassifier`: classify a Unit, this must be after a unit type
- `UnitTypeUnitClassifier`: complex classifier for units like `U12`. This will parse a span and will add two new spans `U` and `12` in the section with the proper classification.


This fixes #71 and need more work for #69

missinglink commented:
This looks very promising 👀

Joxit commented:
I think I will merge this this week and do another PR for #69 if everything is OK :smile:

missinglink commented:
Yeah very nice, could you please update the `mask` functionality?

We use that in `pelias/api`

Joxit commented:
Are you OK with this one ? U for both `unit` and `unit_type` ?

Joxit commented:
Different result order between node 10 and node 12 :man_facepalming: 
Ok, I need to fix this too...

Joxit commented:
Should be ok now :smile:

missinglink commented:
Nice, yeah that all looks great :star:

missinglink commented:
Additional test case `Lot 2, Burrows Avenue, EDMONDSON PARK, NSW, Australia`

Joxit commented:
This new test is passing :+1: 

But if I merge this now, it will fail new test from #88 because there are 4 children `Beethovenova` `641/9` `641` and `9`
```
 address CZEs: Beethovenova 641/9, Brno
    1m✖ should be equivalent
    -----------------------
      operator: deepEqual
      expected: |-
        [ { street: 'Beethovenova' }, { housenumber: '641/9' }, { locality: 'Brno' } ]
      actual: |-
        undefined
      at: Test.<anonymous> (/home/travis/build/pelias/parser/test/common.js:22:9)
      stack: |-
  Error: should be equivalent
  at Test.assert [as _assert] (/home/travis/build/pelias/parser/node_modules/tape/lib/test.js:228:54)
  at Test.bound [as _assert] (/home/travis/build/pelias/parser/node_modules/tape/lib/test.js:80:32)
  at Test.tapeDeepEqual (/home/travis/build/pelias/parser/node_modules/tape/lib/test.js:426:10)
  at Test.bound [as deepEquals] (/home/travis/build/pelias/parser/node_modules/tape/lib/test.js:80:32)
  at Test.<anonymous> (/home/travis/build/pelias/parser/test/common.js:22:9)
  at Test.bound [as _cb] (/home/travis/build/pelias/parser/node_modules/tape/lib/test.js:80:32)
  at Test.run (/home/travis/build/pelias/parser/node_modules/tape/lib/test.js:96:10)
  at Test.bound [as run] (/home/travis/build/pelias/parser/node_modules/tape/lib/test.js:80:32)
  at Immediate.next [as _onImmediate] (/home/travis/build/pelias/parser/node_modules/tape/lib/results.js:81:19)
  at processImmediate (internal/timers.js:456:21)
```

missinglink commented:
We've never had 'words' with overlapping character ranges before:

```
641/9  13:18
641  13:16
9  17:18
```

Is it possible to remove the `641/9` 'word' so that's no longer the case?

Joxit commented:
This was introduced in #56, I named it `Alternative spans`. I added in this PR `/`

Now we should navigate through the graph.

I fixed the test :smile:

missinglink commented:
Cool :+1: we can address the graph relationships in https://github.com/pelias/parser/issues/47
